lm_eval/models/huggingface.py

```diff
@@ -448,7 +469,16 @@ def _get_backend(
         Helper method during initialization.
         Determines the backend ("causal" (decoder-only) or "seq2seq" (encoder-decoder))
         model type to be used.
+        sets `self.AUTO_MODEL_CLASS` appropriately if not already set.
         """
+        # escape hatch: if we're using a subclass that shouldn't follow
+        # the default _get_backend logic,
+        # then skip over the method.
+        # TODO: this seems very much undesirable in some cases--our code in HFLM
```

Comment:
This might be fine for now but we should consider switching in future to setting a `self.backend = "causal"` that we check for equality against in HFLM, rather than the current `self.AUTO_MODEL_CLASS == AutoModelForCausalLM` behavior?